sdk/tests/deqp/framework/opengl/gluDrawUtil.js

```diff
@@ -40,7 +40,7 @@ var gluShaderProgram = framework.opengl.gluShaderProgram;
  */
 gluDrawUtil.VertexArrayBinding = function(type, location, components, elements, data, stride, offset) {
     this.type = type;
-    this.location = location === undefined ? -1 : location;
+    this.location = location === undefined ? 0 : location;
```

Comment:
It would be safest to leave the code as:
this.location = location === undefined ? 0 : location;
since the previous code explicitly tested against "undefined". I don't know how all the callers work.

Looking through gluDrawUtil.js, there may be other suspect pieces of code. gluDrawUtil.bindingPointFromName sets the location to -1 if the location is undefined (I see you modified it below as well), and functions like gluDrawUtil.vabFromBindingPointAndArrayPointer take the BindingPoint and set the location of the new gluDrawUtil.VertexArrayBinding to it.

-1 is the canonical "undefined attribute location" value. For this reason, I wonder whether a more reliable fix would be to change gluDrawUtil.namedBindingsToProgramLocations to check:
  if (cur.location >= 0)
rather than
  if (cur.location)

and leave this and gluDrawUtil.bindingPointFromName alone. What do you think?